processing/src/main/java/org/apache/druid/segment/filter/BoundFilter.java

```diff
@@ -75,71 +76,107 @@ public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
     if (!Filters.checkFilterTuningUseIndex(boundDimFilter.getDimension(), selector, filterTuning)) {
       return null;
     }
-    if (supportShortCircuit()) {
+    if (supportStringShortCircuit()) {
       final ColumnIndexSupplier indexSupplier = selector.getIndexSupplier(boundDimFilter.getDimension());
       if (indexSupplier == null) {
         return Filters.makeNullIndex(doesMatchNull(), selector);
       }
       final LexicographicalRangeIndex rangeIndex = indexSupplier.as(LexicographicalRangeIndex.class);
-      if (rangeIndex == null) {
```

Comment:
was this a bug?

Reply:
it was a bug with single type numeric nested literal columns and potentially future columns that only have some of these index types, but not `STRING` typed columns which have both lexicographical range indexes and predicate indexes.